pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -59,52 +61,64 @@ func (a *index) serviceServiceBuilder(
 				TargetPortName: p.TargetPort.StrVal,
 			}
 		}
-		waypointKey := ""
-		waypoint := fetchWaypointForService(ctx, waypoints, namespaces, s.ObjectMeta)
+		waypointStatus := model.WaypointBindingStatus{}
+		waypoint, wperr := fetchWaypointForService(ctx, waypoints, namespaces, s.ObjectMeta)
 		if waypoint != nil {
-			waypointKey = waypoint.ResourceName()
+			waypointStatus.ResourceName = waypoint.ResourceName()
 		}
+		waypointStatus.Error = wperr
+
 		a.networkUpdateTrigger.MarkDependant(ctx) // Mark we depend on out of band a.Network
 		return &model.ServiceInfo{
 			Service:       a.constructService(s, waypoint),
 			PortNames:     portNames,
 			LabelSelector: model.NewSelector(s.Spec.Selector),
-			Source:        kind.Service,
-			Waypoint:      waypointKey,
+			Source:        MakeSource(s),
+			Waypoint:      waypointStatus,
 		}
 	}
 }
 
+func MakeSource(o controllers.Object) model.TypedObject {
```

Comment:
super nit: seems like the helper should live in model

Reply:
I think model cannot import controllers unfortunately